### DISCUSSION TITLE
uploading larger files into s3 bucket using pre-signed URL's

### DISCUSSION THREAD

**Discussion by ashsyed** [community/community#20397] [Other Feature Feedback, Questions, & Ideas]
[fileupload](https://github.com/awsdocs/aws-doc-sdk-examples/blob/ac0821ea293d75b23b156e5e0e9c4be9bb70a3bc/javav2/example_code/s3/src/main/java/com/example/s3/GeneratePresignedUrlAndUploadObject.java#L97) 

@scmacdon
Is there any standards followed to upload larger files into s3 using pre-signed url's? 
copying larger files into output stream is causing heap space issue.

**Comment by FahdSeddik** [comment]
Hey!
I think you might find an answer in this [link](https://aws.amazon.com/premiumsupport/knowledge-center/s3-upload-large-files/#:~:text=When%20you%20upload%20large%20files,cp%20and%20aws%20s3%20sync.). Hope this helps!

**Reply by ashsyed** [reply]
Hi Fahad, First of all thanks for your prompt response. Currently we are not using AWS CLI instead we are using presigned URLs to upload files because of restrictions. Could you shed some light on the same ? Thank you in advance
